### ISSUE TITLE
scatter tasks can't be re-used

### ISSUE THREAD

**Issue by natechols** [PacificBiosciences/pbsmrtpipe#28]
The scatter-task-id in the chunk operator XML is currently treated as a unique key (especially in bgraph.py).  Since this is being used to look up the corresponding gather task, it can and does end up with the wrong gather, which then results in downstream tasks being passed incorrect file types.  I can work around this by creating a duplicate scatter task with a unique ID (this requires an extra JSON tool contract and a very few lines of Python), but this is a temporary solution at best.

**Comment by evolvedmicrobe** [comment]
Out of curiosity, is this why `scatter_alignments_reference.py` and `scatter_alignments_reference_basemods.py` are different files in pbcoretools?

**Comment by natechols** [comment]
Correct, although in practice we don't need both of those any more, since the basemods pipeline no longer includes GenomicConsensus.
